File: 2024_java_basics/10-for-loops-lab/tasks/06-oscars.java
import java.util.Scanner;

public class Oscars {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String nameActor = scanner.nextLine();
        double pointsAcademy = scanner.nextDouble();
        int numberOfJuries = scanner.nextInt();
        scanner.nextLine(); // Consume the newline character after nextInt()

        for (int i = 1; i <= numberOfJuries; i++) {
            String nameJury = scanner.nextLine();
            double pointsJury = scanner.nextDouble();
            scanner.nextLine(); // Consume the newline character after nextDouble()

            int nameJuryLength = nameJury.length();

            pointsAcademy += ((nameJuryLength * 1.0 * pointsJury) / 2);

            if (pointsAcademy >= 1250.5) {
                break;
            }
        }

        if (pointsAcademy >= 1250.5) {
            System.out.printf("Congratulations, %s got a nominee for leading role with %.1f!%n", nameActor, pointsAcademy);
        } else {
            double pointsNeeded = 1250.5 - pointsAcademy;
            System.out.printf("Sorry, %s you need %.1f more!%n", nameActor, pointsNeeded);
        }

        scanner.close();
    }
}

File: 2022_c#_basics/10-for-loop-exercise/06-oscars.cs
using System;

namespace _06_Oscars
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string nameActor = Console.ReadLine();
            double pointsAcademy = double.Parse(Console.ReadLine());
            int numberOtzenqvasht = int.Parse(Console.ReadLine());

            for (int i = 1; i <= numberOtzenqvasht; i++)
            {
                string nameOtzenqvasht = Console.ReadLine();
                double pointsOtzenqvasht = double.Parse(Console.ReadLine());

                int nameOtzenqvashtLength = nameOtzenqvasht.Length;

                pointsAcademy = pointsAcademy + ((nameOtzenqvashtLength * 1.0 * pointsOtzenqvasht) / 2);

                if (pointsAcademy >= 1250.5)
                {
                    break;
                }

            }

            if (pointsAcademy >= 1250.5)
            {
                Console.WriteLine($"Congratulations, {nameActor} got a nominee for leading role with {pointsAcademy:f1}!");
            }
            else
            {
                double pointsNeeded = 1250.5 - pointsAcademy;
                Console.WriteLine($"Sorry, {nameActor} you need {pointsNeeded:f1} more!");
            }
        }
    }
}
